projects/pswg/src/main/java/com/parzivail/pswg/entity/ThermalDetonatorEntity.java

```diff
@@ -0,0 +1,135 @@
+			world.spawnParticles(player,SwgParticles.EXPLOSION_SMOKE, true, x, y+0.5f, z, 15*m, 0.25*m, 0.125*m, 0.125*m, 0.025);
+			world.spawnParticles(player,SwgParticles.EXPLOSION_SMOKE, true, x, y+0.5f, z, 15*m, 0.125*m, 0.125*m, 0.25*m, 0.02);
+			world.spawnParticles(player,SwgParticles.EXPLOSION_SMOKE, true, x, y+0.5f, z, 10*m, 0.1*m, 0.2*m, 0.1*m, 0.0075);
+			world.spawnParticles(player,SwgParticles.EXPLOSION_SMOKE, true, x, y+0.5f, z, 10*m, 0.2*m, 0.1*m, 0.25*m, 0.01);
+			world.spawnParticles(player, SwgParticles.EXPLOSION_SMOKE, true, x, y+0.5f, z, 10 * m, 0.25 * m , 0.5 * m, 0.3 * m, 0.015);
+
+			world.spawnParticles(player,ParticleTypes.FLAME, true, x, y, z, 20*m, 0.5*m, 0.5*m, 0.5*m, 0.1);
+			world.spawnParticles(player,ParticleTypes.SMALL_FLAME, true, x, y, z, 20*m, 1.2*m, 1.2*m, 1.2*m, 0.125);
+			world.spawnParticles(player,ParticleTypes.SMALL_FLAME, true, x, y, z, 20*m, 1.1*m, 1.3*m, 1.25*m, 0.125);
+		}
+	}
+
+	@Override
+	public boolean canBeHitByProjectile()
+	{
+		return true;
+	}
+
+	@Override
+	protected void onCollision(HitResult hitResult)
+	{
+
+		if (hitResult.getType() == HitResult.Type.BLOCK)
+		{
+			if(hitResult.squaredDistanceTo(this)<0.01){
+				this.setVelocity(0,0,0);
+			}
+			BlockHitResult blockHitResult = (BlockHitResult)hitResult;
+
+			Vec3d velocity = this.getVelocity();
+			Vec3d pos = hitResult.getPos();
+
+			double modX = 0.9f;
+			double modY = 0.9f;
+			double modZ = 0.9f;
+			if(blockHitResult.getSide()==Direction.UP){
+				modY=-0.2f;
+			}else if(blockHitResult.getSide()==Direction.DOWN){
+				modY=-1.2f;
+			}else {
+				if (this.getX() - pos.getX() > 0 && this.getZ() - pos.getZ() > 0)
+				{
+					if (blockHitResult.getSide() == Direction.EAST) {
+						modX = modX * -1;
+					}else {
+						modZ = modZ * -1;
+					}
+				}
+				else if (this.getX() - pos.getX() < 0 && this.getZ() - pos.getZ() > 0)
+				{
+					if (blockHitResult.getSide() == Direction.SOUTH) {
+						modZ = modZ * -1;
+					}else {
+						modX = modX * -1;
+					}
+				}
+				else if (this.getX() - pos.getX() < 0 && this.getZ() - pos.getZ() < 0)
+				{
+					if (blockHitResult.getSide() == Direction.WEST) {
+						modX = modX * -1;
+					}else {
+						modZ = modZ * -1;
+					}
+				}
+				else if (this.getX() - pos.getX() > 0 && this.getZ() - pos.getZ() < 0)
+				{
+					if (blockHitResult.getSide() == Direction.NORTH)
+					{
+						modZ = modZ * -1;
+					}else {
+						modX = modX * -1;
+					}
+				}
+			}
+			this.setVelocity(velocity.x * modX, velocity.y * modY, velocity.z * modZ);
+		}
+		super.onCollision(hitResult);
+	}
+	@Override
+	public boolean canExplosionDestroyBlock(Explosion explosion, BlockView world, BlockPos pos, BlockState state, float explosionPower)
+	{
+		if(state.isIn(SwgTags.Blocks.DETONATOR_EXPLODE)||state.isIn(BlockTags.LEAVES)||state.isIn(BlockTags.CROPS)||state.isIn(BlockTags.SMALL_FLOWERS)||state.isIn(BlockTags.TALL_FLOWERS)||state.isOf(Blocks.GRASS)||state.isOf(Blocks.TALL_GRASS)){
```

Comment:
Let's move everything that isn't `DETONATOR_EXPLODE` to a sub-tag through JSON -- you can add leaves, crops, etc. to `DETONATOR-EXPLODE` tag in TARKIN. Then, you only need one check

Reply:
How would I add leaves, crops, etc to Detonator Explode? Do I just write them individually?